test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceProviderTests.cs

```diff
@@ -2271,6 +2271,34 @@ public void PackageSourcesChanged_EventRunsSubscriptions(bool subscribeToEvent)
             Assert.Equal(subscribeToEvent, eventRun);
         }
 
+        [Fact]
+        public void GetPackageSourceBySource_TwoSourcesWithSameUrl_ReturnsFirstSource()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+
+            const string sourceUrl = "https://contoso.test/nuget/index.json";
+            const string contents = $@"<configuration>
+  <packageSources>
+    <add key=""s1"" value=""{sourceUrl}"" />
```

Comment:
Thoughts on adding a `clear`? I think it would just be as a reminder for the reader that we 1. expect it as best practice and 2. the test doesn't anticipate any other configs to apply. 

```suggestion
    </clear>
    <add key=""s1"" value=""{sourceUrl}"" />
```

Reply:
The file is loaded using an API that loads only the single file. It doesn't do the "normal" directory traversal and multi-file accumulation. It would have been nice if the API name made that clear (even better if it was possible to parse a `Settings` instance from memory, rather than needing a file on disk), but I'm just working what already exists.